qt-bugs@ issue : 178507
Trolltech task ID : 179170 (status: fixed for Qt 4.4.0)
applied: no
author: Matthew Woehlke <mw_triad@users.sourceforge.net>

This patch fixes incorrect setting of the palette color group in
QItemDelegate that resulted in inactive colors being used for widgets
in active windows.

--- a/src/gui/itemviews/qitemdelegate.cpp
+++ b/src/gui/itemviews/qitemdelegate.cpp
@@ -653,10 +653,10 @@
 
     QPen pen = painter->pen();
     QPalette::ColorGroup cg = option.state & QStyle::State_Enabled
-                              ? QPalette::Normal : QPalette::Disabled;
-    if (cg == QPalette::Normal && !(option.state & QStyle::State_Active))
-        cg = QPalette::Inactive;
+                              ? option.palette.currentColorGroup() : QPalette::Disabled;
     if (option.state & QStyle::State_Selected) {
+       if (cg == QPalette::Normal && !(option.state & QStyle::State_Active))
+           cg = QPalette::Inactive;
         painter->fillRect(rect, option.palette.brush(cg, QPalette::Highlight));
         painter->setPen(option.palette.color(cg, QPalette::HighlightedText));
     } else {
